src/main/java/io/stargate/sgv2/jsonapi/service/embedding/gateway/EmbeddingGatewayClient.java

```diff
@@ -156,9 +157,11 @@ else if (value instanceof Boolean)
         .transform(
             resp -> {
               if (resp.hasError()) {
-                throw new JsonApiException(
-                    ErrorCode.valueOf(resp.getError().getErrorCode()),
-                    resp.getError().getErrorMessage());
+                ErrorCode errorCode =
+                    EnumUtils.isValidEnum(ErrorCode.class, resp.getError().getErrorCode())
+                        ? ErrorCode.valueOf(resp.getError().getErrorCode())
+                        : ErrorCode.VECTORIZE_SERVICE_TYPE_UNAVAILABLE;
```

Comment:
Should we log an error if we get unrecognized Enum tho? Or is that not an error case?

Reply:
Just realized this is handled the scenario is handled in EGW. This preventive PR may not be required. Will close this one.

Reply:
Ok.